packages/tocco-ui/src/EditableValue/typeEditors/DateAbstract.js

```diff
@@ -72,24 +72,27 @@ class DateAbstract extends React.Component {
     return null
   }
 
-  componentWillReceiveProps(props) {
-    this.handleButtonVisibility()
-    const locale = this.getLocale(props.intl.locale)
+  componentDidUpdate(prevProps, prevState, snapshot) {
+    const valueDiff = !_isEqual(prevProps.value, this.props.value)
+    const localeDiff = !_isEqual(prevProps.intl.locale, this.props.intl.locale)
+    const altFormatDiff = !_isEqual(
+      prevProps.options.flatpickrOptions.altFormat,
+      this.props.options.flatpickrOptions.altFormat
+    )
+    if (valueDiff || altFormatDiff || localeDiff) {
```

Comment:
i would split it up

`
   if (this.Flatpickr && this.flatpickr) {
 if ( !_isEqual(prevProps.value, this.props.value)) {
 this.flatpickr.setDate(this.props.value, false)
}
...
}
`